pkgs/development/python-modules/scipy/default.nix

```diff
@@ -17,46 +22,156 @@
+    hash = "sha256-5mv6MKK+thoo4rvzCsN5w42x6mreIVKtTIH0JTCOu1I=";
+    # Fetch datasets that are needed for tests
+    postFetch = ''
+      # This directory must stay with this name, as it fits the environment
+      # variable XDG_CACHE_DIR set in `preConfigure`. See more details at:
+      # https://github.com/scipy/scipy/pull/18518#issuecomment-1562350648 And
+      # at: https://github.com/scipy/scipy/pull/17965#issuecomment-1560759962
+      mkdir $out/scipy-data
+    '' + (lib.concatStringsSep "\n" (builtins.map (d:
+      # From some reason, if ssl certificates are not available in this
+      # environment, see also:
+      # https://discourse.nixos.org/t/fetchfromgithub-with-a-postfetch-acts-weird/28444/7
+      "${curl} --insecure https://raw.githubusercontent.com/scipy/dataset-${d}/main/${d}.dat -o $out/scipy-data/${d}.dat"
+    ) datasets));
+    fetchSubmodules = true;
   };
-
   patches = [
-    # These tests require internet connection, currently impossible to disable
-    # them otherwise, see:
-    # https://github.com/scipy/scipy/pull/17965
-    ./disable-datasets-tests.patch
+    # Helps with cross compilation, see https://github.com/scipy/scipy/pull/18167
+    (fetchpatch {
+      url = "https://github.com/scipy/scipy/commit/dd50ac9d98dbb70625333a23e3a90e493228e3be.patch";
+      hash = "sha256-Vf6/hhwu6X5s8KWhq8bUZKtSkdVu/GtEpGtj8Olxe7s=";
+      excludes = [
+        "doc/source/dev/contributor/meson_advanced.rst"
+      ];
+    })
   ];
 
-  nativeBuildInputs = [ cython gfortran meson-python pythran pkg-config wheel ];
+  nativeBuildInputs = [ ninja cython gfortran meson meson-python pythran pkg-config wheel ];
 
   buildInputs = [
-    numpy.blas
+    blas
+    lapack
     pybind11
     pooch
+    xsimd
   ] ++ lib.optionals (pythonOlder "3.9") [
     libxcrypt
   ];
 
-  propagatedBuildInputs = [ numpy ];
+  propagatedBuildInputs = [ numpy' ];
 
   nativeCheckInputs = [ nose pytest pytest-xdist ];
 
   doCheck = !(stdenv.isx86_64 && stdenv.isDarwin);
 
   preConfigure = ''
+    # See comment near src's postFetch
+    export XDG_CACHE_HOME=$PWD
     sed -i '0,/from numpy.distutils.core/s//import setuptools;from numpy.distutils.core/' setup.py
     export NPY_NUM_BUILD_JOBS=$NIX_BUILD_CORES
   '';
 
+  # Mostly a copy of pipBuildHook, only with an additional --config-settings
+  # handling cross compilation of meson being run by meson-python, see:
+  # https://github.com/mesonbuild/meson-python/pull/167
+  buildPhase = ''
+    runHook preBuild
+
+    echo "Running meson setup"
+    mkdir -p build
+    meson setup \
+      --cross-file=${crossFile} \
+      -Dblas=${blasName} \
+      -Dlapack=${lapackName} \
+      build
+
+    echo "Creating a wheel..."
+    mkdir -p dist
+    ${python.pythonForBuild.interpreter} -m pip wheel \
+      --verbose \
+      --no-index \
+      --no-deps \
+      --no-clean \
+      --no-build-isolation \
+      --wheel-dir dist \
+      --config-settings 'builddir=build' \
```

Comment:
this should be supported by adding a `pipBuildFlags` to `pip-build-hook.sh` similar as the `pipInstallFlags` for `pip-install-hook.sh`. All you need then is set that variable and `preBuild` without copying the build phase.

Reply:
> this should be supported by adding a `pipBuildFlags` to `pip-build-hook.sh` similar as the `pipInstallFlags` for `pip-install-hook.sh`. All you need then is set that variable and `preBuild` without copying the build phase.

Sounds good! I can prepare a PR and mark this as a draft in the meantime.

Reply:
On a 2nd thought, I'll remove the commit that changes the buildPhase, and not mark this PR as a draft, so we could get going with the other changes that we agree on.